### ISSUE TITLE
You can only access registered factories/objects this way if they are created asynchronously

### ISSUE THREAD

**Issue by mvrsn** [Milad-Akarie/injectable#241]
While trying to figure out why the injections were not placed in a correct order (meaning I always have to manually move some of them to the correct place after generating) I seemingly out of nowhere I started getting a problem I haven't seen before: `You can only access registered factories/objects this way if they are created asynchronously`

I'm having the same issue with `1.4.1.` and `1.5.0`

My goal is to have a `Preferences` class that abstracts away the logic when reading and writing using `SharedPreferences`, and from what I gather I'm supposed to do something like the example below to set this up

```
@module
abstract class RegisterModule {
  @preResolve
  Future<SharedPreferences> get sharedPreferences => SharedPreferences.getInstance();
}
```

```
@singleton
class Preferences {
  SharedPreferences sharedPreferences;

  Preferences(this.sharedPreferences);

  @factoryMethod
  static Future<Preferences> create(SharedPreferences sharedPreferences) async {
    return Preferences(sharedPreferences);
  }
}
```
(I've tried both with and without the `@factoryMethod` with the same results)

This results in the code below being generated, but for some reason this the app does not run with this
```
final registerModule = _$RegisterModule();
await gh.factoryAsync<_i3.SharedPreferences>(
    () => registerModule.sharedPreferences,
    preResolve: true);
...
gh.singletonAsync<_i33.Preferences>(() async =>
      _i33.Preferences.create(await get.getAsync<_i3.SharedPreferences>()));
```

I can't figure out if I am doing something wrong, or if the generation is not producing the correct result as the generated code for the `RegisterModule` does not look like it does in the [documentation](https://github.com/Milad-Akarie/injectable#pre-resolving-futures)

**Comment by Milad-Akarie** [comment]
@mvrsn  that's a bug in version 1.5.0, please downgrade for now. and make sure you're awaiting the dependencies Initialization function before you resolve your classes

**Comment by Milad-Akarie** [comment]
@mvrsn  This should be fixed in injectable_generator 1.5.0 
please reopen if it's not.
